### PR TITLE
Add lua5.3

### PR DESCRIPTION
**Pull request by iggyvolz**
A couple tasks before this can be merged:

- Mirror needs to be added or removed entirely (the file is not up yet on the mirror)
- Both patches need to be verified to work.  One is labeled to work with 5.3-beta, and one is not marked to work with the 5.3 branch at all.

### CODE REVIEW

**Comment by iggyvolz** [comment]
Also, I'm not sure what was with the rest of the file but I went through and did some sensible replacements (5.2 => 5.3, 5.2.3 => 5.3.0, lua52 => lua53, et cetera).

I took this directly from the lua file on homebrew/homebrew.

**Comment by iggyvolz** [comment]
I'll fix the build failures when I get home - I forgot to audit --strict

File: lua53.rb [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+
+  # completion provided by advanced readline power patch
+  # See http://lua-users.org/wiki/LuaPowerPatches
+  patch do ## TODO: Patch reported to work with 5.3-beta, does it work with 5.3 final?
+    url "http://luajit.org/patches/lua-5.2.0-advanced_readline.patch"
+    sha1 "ca405dbd126bc018980a26c2c766dfb0f82e919e"
+  end if build.with? "completion"
+
+  # sigaction provided by posix signalling power patch
+  patch do ## TODO: Does this patch work with 5.3?
```
Comment (DomT4) [inline lua53.rb]: No dice:

```
/usr/bin/patch -g 0 -f -p1 -i /private/tmp/lua53--patch-EZU9BC/lua-5.2.3-sig_catch.patch
patching file src/lua.c
Hunk #1 FAILED at 99.
Hunk #2 FAILED at 167.
2 out of 2 hunks FAILED -- saving rejects to file src/lua.c.rej
/usr/local/Library/Homebrew/debrew.rb:11:in `raise'
ErrorDuringExecution: Failure while executing: /usr/bin/patch -g 0 -f -p1 -i /private/tmp/lua53--patch-EZU9BC/lua-5.2.3-sig_catch.patch
```

**Comment by iggyvolz** [comment]
Took care of style and removed the second patch.

**Comment by DomT4** [comment]
Thanks. You can probably just remove entirely the chunks commented out btw :).

Some good news, and some bad. Good: There's no Lua51 conflict. Bad: There's quite a lot of Lua52 conflict:

```
Possible conflicting files are:
/usr/local/bin/lua -> /usr/local/Cellar/lua/5.2.3_1/bin/lua
/usr/local/bin/luac -> /usr/local/Cellar/lua/5.2.3_1/bin/luac
/usr/local/include/lauxlib.h -> /usr/local/Cellar/lua/5.2.3_1/include/lauxlib.h
/usr/local/include/lua.h -> /usr/local/Cellar/lua/5.2.3_1/include/lua.h
/usr/local/include/lua.hpp -> /usr/local/Cellar/lua/5.2.3_1/include/lua.hpp
/usr/local/include/luaconf.h -> /usr/local/Cellar/lua/5.2.3_1/include/luaconf.h
/usr/local/include/lualib.h -> /usr/local/Cellar/lua/5.2.3_1/include/lualib.h
/usr/local/share/man/man1/lua.1 -> /usr/local/Cellar/lua/5.2.3_1/share/man/man1/lua.1
/usr/local/share/man/man1/luac.1 -> /usr/local/Cellar/lua/5.2.3_1/share/man/man1/luac.1
/usr/local/lib/liblua.dylib -> /usr/local/Cellar/lua/5.2.3_1/lib/liblua.dylib
/usr/local/lib/pkgconfig/lua.pc -> /usr/local/Cellar/lua/5.2.3_1/lib/pkgconfig/lua.pc
```

**Comment by iggyvolz** [comment]
Not sure what exactly I should remove and what I should change to lua53.

I'll look more into this tomorrow.

**Comment by iggyvolz** [comment]
I must confess I have no idea what I'm doing around the patch.  I'm getting a:
```bash
patch: **** malformed patch at line 22: diff --git a/src/Makefile b/src/Makefile <<<EOF

Error: Failure while executing: patch -g 0 -f -p1
```
(Disclaimer: Running this on a linux box because I'm away from my mac ATM)

I tried adding the patch data in an EOF block but that helped nothing.

Maybe it's because I removed the second patch do block?

**Comment by iggyvolz** [comment]
I believe I had it working now - I manually patched makefile and src/makefile and then ran a git diff>>(homebrew-versions path)/lua53.rb.  If the tests pass I'll squash and rebase. (yes - I meant squash! not stash! on all of the commits (facepalm))

**Comment by iggyvolz** [comment]
Squashed and rebased onto master.  I'm 99% sure this will pass (last time it failed on mavericks due to a git error, probably just a network glitch).

**Comment by iggyvolz** [comment]
All good - the only question that remains now is the mirror.

The official mirror (ref: http://www.lua.org/mirrors.html) is http://www.tecgraf.puc-rio.br/lua/mirror/, a redirect to http://webserver2.tecgraf.puc-rio.br/lua/mirror/.  Making the official mirror http://webserver2.tecgraf.puc-rio.br/lua/mirror/ftp/lua-5.3.0.tar.gz.

Should we ask debian if they're willing to mirror lua53, or use the official lua mirror?  Or is a mirror even needed in the first place?

**Comment by DomT4** [comment]
Give me a bit to look at how to avoid this conflicting with our other Lua formulae. It'd be good to have the default being not conflicting, with an option to not rename the binaries for those who just want to rock ` lua53 ` without the others. I'll start work on figuring out that dance imminently.

**Comment by DomT4** [comment]
Feel free to steal the changes from me. I'm not attached to getting credit for the commit; you did the majority of the work here.

My proposal for side-by-side-by-side-by-side Lua family is [here](https://github.com/DomT4/homebrew/commit/a845d001440621546aecc54a2848e6888d155311). Feel free to poll the maintainers for approval on the changes.

**Comment by iggyvolz** [comment]
Sorry about not getting back on this.  Life has gotten fairly hectic recently and I'm not able to contribute as much as I used to (not that I really contributed much lol).

**Comment by DomT4** [comment]
No worries at all, It's understandable. Thanks for getting the ball rolling on this nonetheless.

**Pull request closed without merging**